### ISSUE TITLE
XliffReader returns wrong offset value

### ISSUE THREAD

**Issue by MarcoPNS** [neos/flow-development-collection#2569]
### Description

I have a Main.xlf with 3 file blocks. 2 blocks are fine. When I add a third one, Neos throw following exception:

````
Fatal error: Uncaught TypeError: Argument 2 passed to Neos\Utility\Arrays::arrayMergeRecursiveOverrule() must be of the type array, null given, called in /Users/marco/Sites/xlifftest/Data/Temporary/Development/Cache/Code/Flow_Object_Classes/Neos_Flow_I18n_Xliff_Service_XliffFileProvider.php on line 164 and defined in /Users/marco/Sites/xlifftest/Packages/Libraries/neos/utility-arrays/Classes/Arrays.php on line 71
````

My Main.xlf looks a bit like this:

````
<?xml version="1.0"?>
<xliff version="1.2" xmlns="urn:oasis:names:tc:xliff:document:1.2">
    <file original="" source-language="en" datatype="plaintext">
        <body>
            <trans-unit id="first">
                <source>First</source>
            </trans-unit>
        </body>
    </file>
    <file original="" source-language="en" datatype="plaintext">
        <body>
            <trans-unit id="second">
                <source>Second</source>
            </trans-unit>
        </body>
    </file>
    <file original="" source-language="en" datatype="plaintext">
        <body>
            <trans-unit id="third">
                <source>Third</source>
            </trans-unit>
        </body>
    </file>
</xliff>
````

I digged a little deeper into that problem and found out that the XliffReader->readFiles returns a wrong offset for the third block. Instead of `$offset = 2`, it returns `$offset = 3`. 

### Steps to Reproduce

1. Have a translation file with 3 file blocks
2. Try to output the third value.

#### Expected behavior

Translation get shown correctly

#### Actual behavior

````
Fatal error: Uncaught TypeError: Argument 2 passed to Neos\Utility\Arrays::arrayMergeRecursiveOverrule() must be of the type array, null given, called in /Users/marco/Sites/xlifftest/Data/Temporary/Development/Cache/Code/Flow_Object_Classes/Neos_Flow_I18n_Xliff_Service_XliffFileProvider.php on line 164 and defined in /Users/marco/Sites/xlifftest/Packages/Libraries/neos/utility-arrays/Classes/Arrays.php on line 71
````

### My stupid solution that somehow work

I guess this is a really unsafe way to do it (that's why I only open a issue and not a PR) but currently that works with an endless amount of file blocks.

Replacing following line with following code:

https://github.com/neos/flow-development-collection/blob/403b8c643bf34f1ebc14ee1037d244e11bc3c7e5/Neos.Flow/Classes/I18n/Xliff/Service/XliffReader.php#L54

````
if ($reader->name == 'xliff' || $reader->name == '#text') {
    $offset++;
}
````

### Affected Versions

Neos: 7.1.1

Flow: 7.1.3

**Comment by MarcoPNS** [comment]
Created a PR with a saver solution. See #2570

**Comment by sorenmalling** [comment]
Closed by #2596
